File: app/src/main/java/com/example/dandelion/ui/DetailedPostViewModel.java
package com.example.dandelion.ui;

import android.os.Build;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.annotation.RequiresApi;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.dandelion.instance.Post;
import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.firestore.QueryDocumentSnapshot;
import com.google.firebase.firestore.QuerySnapshot;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import static android.content.ContentValues.TAG;

public class DetailedPostViewModel extends ViewModel {

    private MutableLiveData<Post> post;
    private List<Post> postList;
    private FirebaseAuth mAuth;
    private FirebaseFirestore mFirestore;
    private String uid;

    public DetailedPostViewModel() {
        post = new MutableLiveData<>();
        mAuth = FirebaseAuth.getInstance();
        uid = mAuth.getUid();
        postList = new ArrayList<>();
        mFirestore = FirebaseFirestore.getInstance();
    }


    public LiveData<Post> getObservablePost() {
        return post;
    }



    public void loadPost(String pid) {
        mFirestore.collection("posts")
                .document(pid)
                .get()
                .addOnSuccessListener(new OnSuccessListener<DocumentSnapshot>() {
                    @RequiresApi(api = Build.VERSION_CODES.KITKAT)
                    @Override
                    public void onSuccess(DocumentSnapshot documentSnapshot) {
                            Post p = documentSnapshot.toObject(Post.class);
                            post.setValue(p);
                    }
                });
    }


    //todo load comment

}

File: app/src/main/java/com/example/unlone/ui/lounge/common/DetailedPostViewModel.kt
package com.example.unlone.ui.lounge.common

import android.util.Log
import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData
import com.example.unlone.instance.Post
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import androidx.lifecycle.LiveData
import com.google.firebase.firestore.ktx.toObject
import java.util.ArrayList

class DetailedPostViewModel : ViewModel() {
    private val post: MutableLiveData<Post?> = MutableLiveData()
    private val postList: List<Post>
    private val mAuth: FirebaseAuth = FirebaseAuth.getInstance()
    private val mFirestore: FirebaseFirestore
    private val uid: String? = mAuth.uid
    val observablePost: LiveData<Post?>
        get() = post

    fun loadPost(pid: String) {
        mFirestore.collection("posts")
            .document(pid)
            .get()
            .addOnSuccessListener { documentSnapshot ->
                val p = documentSnapshot.toObject<Post>()
                post.value = p
                Log.d("TAG", "detailedPost: ${post.value.toString()}")
                Log.d("TAG", "pid: $pid")
            }
    } //todo load comment

    init {
        postList = ArrayList()
        mFirestore = FirebaseFirestore.getInstance()
    }
}